### ISSUE TITLE
Issues attempting to use conditional logic

### ISSUE THREAD

**Issue by starlightknight** [Draggable/formeo#215]
Hi Kevin!

First of all, thanks for all your hard work on the widget :-) 

I'm attempting to use the Formeo demo to construct a simple form to demo the conditional logic functionality, and I'm currently a bit confused. 

I made a 4 simple question form - as shown below. For this form, I was using the latest formeo demo page at the time of writing.

![Simple Form](https://i.imgur.com/r8ogBav.png)

![Simple Form Options](https://imgur.com/rOFssTk.png)

The idea is that this would demonstrate extremely primitive conditional logic via hiding the corresponding question based upon the radio button selection.

When I go to insert conditions, based on the question, id expect to see:

```
If (field) value equals (field's value)
Then (target field) is not visible.
```

However, for picking the value, I see only fields in the dropdown, which is also visually glitched (see screenshot, firefox/chrome on linux).

![Form Conditions](https://imgur.com/oavgjyG.png)

Therefore, what it looks like to me is,

```
If (field a) value equals (field b) value <- and then no field appears to pick an actual value from the question
Then (target field) is not visible
```

Which doesn't seem to make sense to me. I would want to show or hide based on a field's value, not two different fields being equal (No Text, No Select, No Number).

I see that the value field allows you to type in free text instead of selecting a field. But if I type the value I assigned in the screenshot, then a crash occurs on render with a nullpointerexception.

Am I doing something wrong? How would I accomplish this use case? 

Other things I noticed while creating this form:

- If inline-edit the labels on the question, this does not seem to synchornize with the values shown under options if I edit it.
- The dropdowns for conditional logic do not expand appropriately so options are cut off and its hard to scroll. it looks like they cannot drop down further than the panel theyre in.
- After selecting an option from the conditional logic drop down, if i wish to change it, the drop down does not appear again when I click it unless I clear the field first, which is cumbersome as a user.
- A text area can expand outside of the panel that contains it by using the standard html5 resize.
- I see that you've factored out the need to provide a name for the field and are just using the label in the conditional dropdowns. This is really good as many other form builders such as google forms do the same. But I'm wondering why you did not do the same with values. It seems like you should be able to have the value correlate to a generated uuid as well, so that you don't have to ask the user to provide one. Then you could just use the value label like you do with the question label.

**Comment by kevinchappell** [comment]
Agreed, conditions needs work. In fact in heavy use it's apparent that the conditions tab should only have a plain text summary of the condition similar to your explanation with an edit button that launches a modal to actually edit the condition. The new form will solve layout and overflow issues.

>If inline-edit the labels on the question, this does not seem to synchornize with the values shown under options if I edit it.

Issue: #218

>The dropdowns for conditional logic do not expand appropriately so options are cut off and its hard to scroll. it looks like they cannot drop down further than the panel theyre in.

Will be fixed with new UI.

> After selecting an option from the conditional logic drop down, if i wish to change it, the drop down does not appear again when I click it unless I clear the field first, which is cumbersome as a user.

If I understand correctly, when a user changes "equals" to "contains" is it expected that the "target/value" should clear?

> A text area can expand outside of the panel that contains it by using the standard html5 resize.

Issue: #217 

> I see that you've factored out the need to provide a name for the field and are just using the label in the conditional dropdowns. This is really good as many other form builders such as google forms do the same. But I'm wondering why you did not do the same with values. It seems like you should be able to have the value correlate to a generated uuid as well, so that you don't have to ask the user to provide one. Then you could just use the value label like you do with the question label.

I'm not sure I fully understand the problem. If a comparison field is selected a dropdown to choose whether to compare its value or visibility should appear. Users can compare arbitrary values by not selecting a comparison field. Can you share your formData so I may be able to replicate?

![image](https://user-images.githubusercontent.com/1457540/59065647-0b580080-8862-11e9-8a0d-3b8a95f7291c.png)

**Comment by starlightknight** [comment]
@kevinchappell I apologize for the delayed response, I was traveling this weekend :-)

Let me see if I can explain better via a series of screenshots...

Let's start with how things were in FormBuilder...
Here, you had to pick both names and labels for each options.

![FormBuilder](https://i.imgur.com/bqmlHkJ.png)

In Formeo, name/label are combined for a field... but not a value

![FormeoLabels](https://i.imgur.com/cLJjobO.png)

In other form services, like Google Forms, FormAssembly, or others - the concept of "values" are never asked. Everything is just the label.

![GoogleForms](https://i.imgur.com/V0euNTR.png)

So the feedback I always get from users when exposing Formeo as a widget is, "why do we have to enter a value here? I don't know what to put, or why it matters".

Entering into conditionals, they get more confused... 

![Conditions1](https://i.imgur.com/KvFUqmB.png)

![Conditions2](https://i.imgur.com/F0UBFpV.png)

![Conditions3](https://i.imgur.com/wHJ3TmB.png)

Does this convey my question better?

Thanks again for your hard work on this widget. It looks very promising :-)

**Comment by shaunakv1** [comment]
@kevinchappell  Any update on this issue? Looks like this is still broken.
